pyart/retrieve/kdp_proc.py

```diff
@@ -21,11 +33,994 @@
+    f_transposed = f.T
+
+    # Measurement prediction matrix--------------------------------
+    #J. Grazioli modification 07.2015 --previous H_plus buggy--
+    h_plus = np.array([[-2*dr,1,0,1],[2*dr,1,1,0],[0,-1,0,0]],dtype=float)
+   
+      
+    #Define the input
+    psidp = psidp_in
+    #Get indices of finite data
+    real_data_ind = np.where(np.isfinite(psidp.ravel()))[0]    
+    offset = real_data_ind[0]
+    
+    if len(real_data_ind):   
+       mpsidp = psidp.ravel()[real_data_ind[-1]]
+    else:   
+       mpsidp = np.nan
+   
+    psidp = psidp[offset:real_data_ind[-1]+1]   
+ 
+    nrg = len(psidp)
+
+    #Define the output
+    kdp_filter_out = np.zeros([nrg,])
+   
+    kdp_mat = np.zeros([nrg,2*len(SCALERS)])
+    kdp_sim = np.zeros([nrg,len(SCALERS)])
+    phidp_filter_out = np.zeros([nrg])
+    
+    #==================================================================
+    #Prepare a longer array with some extra gates on each side
+   
+    #add  values at the beginning and at the end of the profile
+    psidp_long = np.zeros([nrg + PADDING * 2,]) * np.nan
+   
+    nn = nrg + PADDING * 2
+
+    noise = 2*np.random.randn(PADDING)
+    psidp_long[0:PADDING] = noise + psidp[0]
+    psidp_long[nrg + PADDING: nrg+ 2*PADDING] = mpsidp + noise
+    psidp_long[PADDING:nrg + PADDING] = psidp
+
+    psidp = psidp_long
+    
+    #Get information of valid and non valid points in psidp the new psidp
+    nonan = np.where(np.isfinite(psidp))[0]
+    nan =  np.where(np.isnan(psidp))[0]
+   
+    ranged = np.arange(0,nn)
+    
+    psidp_interp = psidp
+
+    # interpolate
+    if len(nan):
+        interp = interpolate.interp1d(ranged[nonan],psidp[nonan],kind='zero')
+        psidp_interp[nan] = interp(ranged[nan])
+    else:
+        psidp_interp = psidp
+       
+    # add noise
+    if len(nan):   
+       psidp_interp[nan] = psidp_interp[nan] + 2 * np.random.randn(len(nan))
+   
+    #Define the final input and output
+    psidp = psidp_interp
+
+    #--------------------Smallest scaler------------------------
+    scaler = 10 ** (-2.)
+
+
+    #Backward
+    kdp_dummy_b2,error_kdp = _kdp_estimation_backward_fixed(psidp, rcov, 
+                          pcov * scaler, f, f_transposed, h_plus, c1, c2, 
+                          b1, b2, kdp_th, mpsidp)
+    kdp002 = kdp_dummy_b2[PADDING:nrg + PADDING]
+   
+    #Forward
+    kdp_dummy_b2,phidp_dummy_b2,error_kdp  = _kdp_estimation_forward_fixed(psidp, 
+            rcov, pcov * scaler, f, f_transposed, h_plus, c1, c2,
+            b1, b2, kdp_th)
+    kdp002f = kdp_dummy_b2[PADDING:nrg + PADDING]
+    
+    #Generate the ensemble of Kalman filters estimates in backward and
+    #Forward directions    
+    #--------------------Loop on scalers-------------------------------
+   
+    for i, sc in enumerate(SCALERS):     
+        #Forward
+        kdp_dummy_f2,phidp_dummy_f2, error_kdp = _kdp_estimation_forward_fixed(psidp,
```

Comment:
Use _ for the dummy arguments which are not used, e.g. `kdp_dummy_f2, _, error_kdp = _kdp_estimation_forward_fixed(`.  Also error_kdp is not used in this function.